com.unity.ml-agents/Runtime/Demonstrations/DemonstrationRecorder.cs

```diff
@@ -0,0 +1,161 @@
+using System.IO.Abstractions;
+using System.Text.RegularExpressions;
+using UnityEngine;
+using System.Collections.Generic;
+using System.IO;
+using MLAgents.Sensor;
+
+namespace MLAgents
+{
+    /// <summary>
+    /// Demonstration Recorder Component.
+    /// </summary>
+    [RequireComponent(typeof(BehaviorParameters))]
+    [AddComponentMenu("ML Agents/Demonstration Recorder", (int)MenuGroup.Default)]
+    public class DemonstrationRecorder : MonoBehaviour
+    {
+        [Tooltip("Whether or not to record demonstrations.")]
+        public bool record;
+
+        [Tooltip("Base demonstration file name. Will have numbers appended to make unique.")]
+        public string demonstrationName;
+
+        [Tooltip("Base directory to write the demo files. If null, will use {Application.dataPath}/Demonstrations.")]
+        public string demoDirectory;
+
+        DemonstrationStore m_DemoStore;
+        public const int MaxNameLength = 16;
```

Comment:
public or internal?

Reply:
this was public before (github doesn't recognize it as move). internal is fine though.